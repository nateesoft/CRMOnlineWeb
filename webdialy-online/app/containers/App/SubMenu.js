import React, { useState } from 'react';
import { FormattedMessage } from 'react-intl';
import { makeStyles } from '@material-ui/core/styles';
import CardGiftcardIcon from '@material-ui/icons/CardGiftcard';
import LocalMallIcon from '@material-ui/icons/LocalMall';
import RecentActorsIcon from '@material-ui/icons/RecentActors';
import BottomNavigation from '@material-ui/core/BottomNavigation';
import BottomNavigationAction from '@material-ui/core/BottomNavigationAction';
import history from 'utils/history';
import * as path from './constants';
import { scope } from './messages';

const useStyles = makeStyles({
  root: {
    width: '100%',
    position: 'fixed',
    bottom: 0,
    zIndex: 999,
  },
});

const SubMenu = props => {
  const classes = useStyles();
  const [value, setValue] = useState(0);
  const handleClick = path => history.push(path);

  if (props.profile.member_role !== 'member') {
    return <span />;
  }
  return (
    <BottomNavigation
      value={value}
      onChange={(event, newValue) => {
        setValue(newValue);
        if (newValue === 0) {
        }
      }}
      showLabels
      className={classes.root}
    >
      {props.leftMenu &&
        props.leftMenu.map(item => {
          let icon = null;
          if (item.icon === 'CardGiftcardIcon') {
            icon = <CardGiftcardIcon />;
          } else if (item.icon === 'LocalMallIcon') {
            icon = <LocalMallIcon />;
          } else if (item.icon === 'RecentActorsIcon') {
            icon = <RecentActorsIcon />;
          }
          return (
            <BottomNavigationAction
              key={item.id}
              icon={icon}
              label={<FormattedMessage id={`${scope}.menu${item.id}`} />}
              style={{
                border: '1px solid chocolate',
                borderRadius: '10px 10px 5px 5px',
                margin: '1px',
              }}
              onClick={() => handleClick(path.publicPath + item.to_path)}
            />
          );
        })}
    </BottomNavigation>
  );
};

export default SubMenu;
