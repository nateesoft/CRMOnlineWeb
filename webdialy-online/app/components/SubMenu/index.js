import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import { makeStyles } from '@material-ui/core/styles';
import CardGiftcardIcon from '@material-ui/icons/CardGiftcard';
import LocalMallIcon from '@material-ui/icons/LocalMall';
import RecentActorsIcon from '@material-ui/icons/RecentActors';
import BottomNavigation from '@material-ui/core/BottomNavigation';
import BottomNavigationAction from '@material-ui/core/BottomNavigationAction';
import history from 'utils/history';
import * as path from 'containers/App/constants';
import { scope } from 'containers/App/messages';

const useStyles = makeStyles({
  root: {
    width: '100%',
    bottom: 5,
    zIndex: 999,
    boxShadow: '1px 2px #aaa',
    borderRadius: '5px 5px 5px 5px',
    position: 'fixed',
  },
  bottom: {
    borderRadius: '10px 10px 5px 5px',
    background: 'none',
    margin: '1px',
  },
});

const SubMenu = props => {
  const classes = useStyles();
  const [value, setValue] = useState(0);
  const handleClick = p => history.push(p);

  const { profile } = props;

  if (profile && profile.member_role) {
    if (profile.member_role !== 'member') {
      return null;
    }
  }
  if (profile && Object.keys(profile).length === 0) {
    return null;
  }

  return (
    <BottomNavigation
      value={value}
      onChange={(event, newValue) => {
        setValue(newValue);
      }}
      showLabels
      className={classes.root}
      style={{ position: props.title === 'Shopping' ? 'relative' : 'fixed' }}
    >
      {props.leftMenu &&
        props.leftMenu.map(item => {
          let icon = null;
          if (item.icon === 'CardGiftcardIcon') {
            icon = <CardGiftcardIcon />;
          } else if (item.icon === 'LocalMallIcon') {
            icon = <LocalMallIcon />;
          } else if (item.icon === 'RecentActorsIcon') {
            icon = <RecentActorsIcon />;
          }
          return (
            <BottomNavigationAction
              key={item.id}
              icon={icon}
              label={<FormattedMessage id={`${scope}.menu${item.id}`} />}
              className={classes.bottom}
              onClick={() => handleClick(path.publicPath + item.to_path)}
            />
          );
        })}
    </BottomNavigation>
  );
};

SubMenu.propTypes = {
  profile: PropTypes.object,
  title: PropTypes.string,
  leftMenu: PropTypes.array,
};

export default SubMenu;
