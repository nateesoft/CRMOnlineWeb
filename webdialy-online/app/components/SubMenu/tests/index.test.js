import React from 'react';
import renderer from 'react-test-renderer';

import SubMenu from '../index';

describe('SubMenu components', () => {
  it('test snapshot', () => {
    const component = renderer.create(<SubMenu />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
