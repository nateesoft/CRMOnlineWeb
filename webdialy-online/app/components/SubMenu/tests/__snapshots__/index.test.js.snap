// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubMenu components test snapshot 1`] = `
<div
  className="MuiBottomNavigation-root makeStyles-layoutFixed-4"
/>
`;
